core/msg.c

```diff
@@ -269,11 +268,11 @@ int msg_send_bus(msg_t *m, msg_bus_t *bus)
 
 int msg_send_receive(msg_t *m, msg_t *reply, kernel_pid_t target_pid)
 {
-    assert(sched_active_pid != target_pid);
+    assert(thread_getpid() != target_pid);
     unsigned state = irq_disable();
-    thread_t *me = (thread_t *)sched_threads[sched_active_pid];
+    thread_t *me = thread_get_active();
```

Comment:
This is the reason

Reply:
I see a small increase in performance with tests/bench_msg_pingpong on the samr21-xpro.
from:
```
{ "result" : 60075 }
```
To (with this PR)
```
{ "result" : 60683 }
```